pkg/eventgenerator/http/generator.go

```diff
@@ -0,0 +1,92 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Generator struct {
+	clientset *kubernetes.Clientset
+	config    *rest.Config
+	logger    logger.Logger
+	namespace string
+	podName   string
+}
+
+func NewGenerator(config *rest.Config, log logger.Logger) (*Generator, error) {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %v", err)
+	}
+	return &Generator{
+		clientset: clientset,
+		config:    config,
+		logger:    log,
+		namespace: "default",
+		podName:   fmt.Sprintf("http-eventgen-pod-%d", time.Now().Unix()),
+	}, nil
+}
+
+func (h *Generator) Generate(params map[string]string, count int, interval time.Duration) (error) {
+	pods, err := h.clientset.CoreV1().Pods(h.namespace).List(context.Background(), metav1.ListOptions{})
+    if err != nil {
+        return fmt.Errorf("failed to list pods: %v", err)
+    }
+
+    // Check for any pod with the dns-eventgen-pod prefix
+    for _, pod := range pods.Items {
+        if strings.HasPrefix(pod.Name, "http-eventgen-pod-") {
+            return fmt.Errorf("HTTP event generator is already running with pod %s, please stop it first", pod.Name)
+        }
+    }
+	
+	url, ok := params["url"]
+	if !ok {
+		return fmt.Errorf("url parameter is required for HTTP event generator")
+	}
+	sleepCount := fmt.Sprintf("%.0fs", interval.Seconds())
```

Comment:
`sleepCount` variable is introduced as when I give `--eventgen-interval=1m`. time.Duration parses it as 1m0s but it is not compatible with `sleep` command. it needs `sleep 1m 0s`, so either we change it and then pass to sleep command and we have to do this with `m` and `h` separately.
Instead I am changing everything in seconds and then passing it to `sleep` command.